[DOCS] Adds the 8.0.0-rc1 release notes (#120806)
elastic/kibana: docs/CHANGELOG.asciidoc

[float]
[[breaking-changes-8.0.0-rc1]]
=== Breaking changes

Breaking changes can prevent your application from optimal operation and performance.
Before you upgrade, review the breaking changes, then mitigate the impact to your application.

// tag::notable-breaking-changes[]

[discrete]
[[breaking-118854]]
.Splits package policy `upgrade` endpoint for Fleet
[%collapsible]
====
*Details* +
For package policy upgrades, the packagePolicy `upgrade` endpoint format supports a mutative upgrade operation (when `dryRun: false`) and a read-only dry run operation (when `dryRun: true`):

[source,text]
--
 POST /package_policies/upgrade
 {
   packagePolicyIds: [...],
   dryRun: false
 }
--

For more information, refer to {kibana-pull}118854[#118854].

*Impact* +
The endpoint is now split into two separate endpoints:

[source,text]
--
 POST /package_policies/upgrade
 {
   packagePolicyIds: [...]
 }

 POST /package_policies/upgrade/dry_run
 {
   packagePolicyIds: [...]
 }
--
====
      
[discrete]
[[breaking-119945]]
.Removes APM jobs from Machine Learning
[%collapsible]
====
*Details* +

> APM NodeJS and RUM Javascript anomaly detection job modules have been removed. For more information, refer to {kibana-pull}119945[#119945].

Reviewer: nit: should be capitalized as JavaScript

Reviewer 2: Node.js too:

```suggestion
APM Node.js and RUM JavaScript anomaly detection job modules have been removed. For more information, refer to {kibana-pull}119945[#119945].
```

*Impact* +

> When you upgrade to 8.0.0, you are unable to create and view the APM NodeJS and RUM Javascript jobs in Machine Learning.

Reviewer: nit: should be capitalized as JavaScript

Reviewer 2: ```suggestion
When you upgrade to 8.0.0, you are unable to create and view the APM Node.js and RUM JavaScript jobs in Machine Learning.
```

====

[discrete]
[[breaking-118300]]
.Fails migrations for unknown types
[%collapsible]
====
*Details* +

> Unknown saved objects types now cause {kib} migrations to fail. For more information, refer to {kibana-pull}118300[#118300].

Reviewer: I'm not sure if this should be "saved objects types" or "saved object types". The latter sounds more correct to me, but I don't know what the rules are 🙂

Reviewer: Also, instead of that PR, I think we should instead link to this issue, which contains much more context on the change: https://github.com/elastic/kibana/issues/107678 

The PR has very little info; most of the discussion that would be relevant to users is in the issue.

*Impact* +
To complete the migration, re enable plugins or delete documents from the index in the previous version.
====

[discrete]
[[breaking-116821]]

> .Removes deprecated config fields from Logs and Metrics APIs and SavedObjects

Reviewer: Missing space: `SavedObjects` -> `Saved Objects`

* `tiebreaker`
* `container`
* `pod`
* `host`

For more information, refer to {kibana-pull}116821[#116821] and {kibana-pull}115874[#115874].

*Impact* +
When you upgrade to 8.0.0, you are unable to use references to the deprecated fields.
====

// end::notable-breaking-changes[]

To review the breaking changes in previous versions, refer to the following: 

<<breaking-changes-8.0.0-beta1,8.0.0-beta1>> | <<breaking-changes-8.0.0-alpha2,8.0.0-alpha2>> | 
<<breaking-changes-8.0.0-alpha1,8.0.0-alpha1>>

[float]
[[deprecations-8.0.0-rc1]]
=== Deprecations

Deprecated functionality does not have an immediate impact on your application, but we strongly recommend you make the necessary updates after you complete the upgrade.

[discrete]
[[deprecation-120689]]
.Renames the `autocreate` data view APM setting
[%collapsible]
====
*Details* +
The `xpack.apm.autocreateApmIndexPattern` APM setting has been removed. For more information, refer to {kibana-pull}120689[#120689].

*Impact* +
To automatically create data views in APM, use `xpack.apm.autoCreateApmDataView`.
====
      
[discrete]
[[deprecation-119494]]
.Updates Fleet API responses for consistency
[%collapsible]
====
*Details* +

> To make sure all Fleet API GET resposes return `items`, the following have been updated:

Reviewer 3: This description doesn't really capture everything that has changed (as described in the impact section). Plus I don't see any detail about `/api/fleet/agents` in under "impact." It feels like the "Details" section is too specific and the "Impact" section is incomplete.

Maybe someone on the dev team should clarify based on what we want to highlight for users. I don't think I'm the person to make this call.

Author: @dedemorton can you add someone from the Dev team to review?

Reviewer 3: I think @juliaElastic is probably the best person to comment since they opened https://github.com/elastic/kibana/pull/119494.

Reviewer 4: @dedemorton hi, is the CHANGELOG generated from somewhere or are you expecting me to suggest how to update it?

Reviewer 3: @juliaElastic I think these entries begin in the changelog, but get edited and formatted by the Kibana writers. You should suggest changes if you think (as I do) that this section needs more work.

Author: @dedemorton @juliaElastic I need to merge these release notes in preparation for the release tomorrow. Please feel free to open a separate PR for any changes you want to make, and let me know if you need any help.

Reviewer 4: @dedemorton 
So to summarize: 
Changes in Fleet API for consistency, previous versions are still working, and deprecated.

1. Rename hyphens to undescores:
- /api/fleet/enrollment-api-keys
- /api/fleet/agent-status
- /api/fleet/service-tokens
2. Change packages endpoint to split `pkgkey` path parameter:
- /epm/packages/{pkgkey} to /epm/packages/{packageName}/{version}
3. Rename `response` or `list` to `items` in response properties:
-  /api/fleet/enrollment_api_keys
- /api/fleet/agents
- /epm/packages/
- /epm/categories
- /epm/packages/_bulk
- /epm/packages/limited
- POST and DELETE /epm/packages/{packageName}/{version}
4. Rename `response` to `item` in response properties:
- GET and PUT /epm/packages/{packageName}/{version}

* Errors: Enhancements to the Errors list page (part II) {kibana-pull}118878[#118878]
* Store Alerts View table state in localStorage {kibana-pull}118207[#118207]
* Handle other values popup when correlated value is not in top 10 {kibana-pull}118069[#118069]
* Adds links to navigate from alerts table to rule {kibana-pull}118035[#118035]
* Reinstates ML multi-metric job {kibana-pull}117836[#117836]
* Re-enables metric-based UI {kibana-pull}117021[#117021]
* Make Alerts page use shared {kib} time range {kibana-pull}115192[#115192]
* Adds enabled toggle {kibana-pull}119994[#119994]
* Adds missing tooltip to the report metric badge in *Exploratory View* {kibana-pull}119940[#119940]
* Adds step duration in step list {kibana-pull}116266[#116266]
Platform::
Moves developer architecture docs to user docs {kibana-pull}119125[#119125]
Reporting::
* Decouples screenshotting plugin from the reporting {kibana-pull}120110[#120110]
* Updates the design of the *Reports* management UI, including the addition of a link to {kib} app where the report was generated {kibana-pull}111412[#111412]
Security::
Adds ability to clone role mappings {kibana-pull}118434[#118434]

> 

Reviewer 5: We have one more that can be added for the Security team:
```suggestion
Adds user logout audit events {kibana-pull}121455[#121455]

```

* Adds Agent key table in APM settings {kibana-pull}119543[#119543]
* Allows users to set Download Speed, Upload Speed, and Latency for their synthetic monitors {kibana-pull}118594[#118594]

[[enhancements-and-bug-fixes-v8.0.0-rc1]]
=== Enhancements and bug fixes

For detailed information about the 8.0.0-rc1 release, review the enhancements and bug fixes.

[float]
[[enhancement-v8.0.0-rc1]]
=== Enhancements
Canvas::
Reverts By-Value Embeddables {kibana-pull}117613[#117613]
Discover::
Adds multi-layer time axis for opt-out only {kibana-pull}115853[#115853]
Elastic Security::
For the Elastic Security 8.0.0 release information, refer to {security-guide}/release-notes.html[_Elastic Security Solution Release Notes_].
Fleet::
Adds consistent `_meta` property to all Fleet ES assets {kibana-pull}119380[#119380]
Kibana Home & Add Data::
Moves overview page and link to the accordion solution title {kibana-pull}114018[#114018]
Lens & Visualizations::
* Handle removal of deprecated date histogram interval in *Vega* {kibana-pull}109090[#109090]
* Adds value labels to Heatmap in *Lens* {kibana-pull}106406[#106406]
Machine Learning::

> Adds support for `force` stop deployment {kibana-pull}118563[#118563]

Reviewer 6: ```suggestion
* Adds support for `force` stop deployment {kibana-pull}118563[#118563]
```

Dashboard::
* Allow text wrapping for panel titles and dashboard descriptions for PDF generation {kibana-pull}121360[#121360]
* Page now resets to zero when rows per page is changed on *Add from Library* window {kibana-pull}118627[#118627]
* Fixes full screen error when pressing back arrow in browser {kibana-pull}118113[#118113]
Elastic Security::
For the Elastic Security 8.0.0 release information, refer to {security-guide}/release-notes.html[_Elastic Security Solution Release Notes_].
Fleet::
* Adds `installed_kibana_space_id` to `epm-packages` saved objects {kibana-pull}120517[#120517]
* Handle Saved Object ID changes {kibana-pull}119527[#119527]
* Fixes upgradeable agents filter {kibana-pull}119338[#119338]
Lens & Visualizations::
* Enables normal mode for percentage charts in *Lens* {kibana-pull}120197[#120197]
* Fixes existing fields query for epoch_millis dates in *Lens* {kibana-pull}119508[#119508]
* Include frozen indices in *Lens* {kibana-pull}118555[#118555]
* Fixes focus on legend action popovers in *Lens* {kibana-pull}115066[#115066]
Machine Learning::
* Fixes data view search based on title {kibana-pull}120737[#120737]

> * Data frame analytics wizard: Only allow data view creation if job will be started {kibana-pull}120042[#120042]

Reviewer 6: ```suggestion
* Data frame analytics wizard: Only allow data view creation if job will be started immediately {kibana-pull}120042[#120042]
```

* Fixes anomaly detection module manifest queries to ignore frozen and cold data tiers {kibana-pull}119635[#119635]

> * Catching error for syntax error in job wizard editor {kibana-pull}119457[#119457]

Reviewer 6: ```suggestion
* Catches syntax error in job wizard editor {kibana-pull}119457[#119457]
```

* Fixes error handling for missing data view in data frame analytics wizard {kibana-pull}119455[#119455]

> * Data Frame Analytics list - ensure refresh works as expected {kibana-pull}117959[#117959]

Reviewer 6: ```suggestion
*  Ensures refresh works as expected in data frame analytics list {kibana-pull}117959[#117959]
```

Reviewer 7: Or `Ensures auto refresh interval is used in data frame analytics list` would provide further clarification.

> * Set data recognizer to ignore frozen indices {kibana-pull}117208[#117208]

Reviewer 6: ```suggestion
* Ignores frozen indices in data recognizer {kibana-pull}117208[#117208]
```

> * Data view loading refactor {kibana-pull}116455[#116455]

Reviewer 6: ```suggestion
* Refactors data view loading {kibana-pull}116455[#116455]
```

Reviewer 7: Can this be moved into the enhancements section (I have edited the label on the PR), and the description amended to `Refactors data view loading to remove unnecessary searches` to provide extra clarification?
